### PR TITLE
Share Thrift Metastore data statistic support across identities

### PR DESCRIPTION
**Pull request by dain**
## Documentation

(X) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(X) Release notes entries required with the following suggested text:

```markdown
# Hive
* Reduce Thrift Metastore communication overhead when impersonation is enabled. ({issue}`13606`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java [hunk 1]
```diff
@@ -171,14 +170,12 @@
     private final boolean translateHiveViews;
     private final boolean assumeCanonicalPartitionKeys;
     private final ThriftMetastoreStats stats;
+    private final MetastoreSupportsDateStatistics metastoreSupportsDateStatistics;
 
     private final AtomicInteger chosenGetTableAlternative = new AtomicInteger(Integer.MAX_VALUE);
     private final AtomicInteger chosenTableParamAlternative = new AtomicInteger(Integer.MAX_VALUE);
     private final AtomicInteger chosesGetAllViewsAlternative = new AtomicInteger(Integer.MAX_VALUE);
 
-    private final AtomicReference<Optional<Boolean>> metastoreSupportsDateStatistics = new AtomicReference<>(Optional.empty());
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java:179]: We need to move chosenGetTableAlternative, chosenTableParamAlternative and chosesGetAllViewsAlternative as well

**Comment by guyco33** [comment]
I approve that `metastoreSupportsDateStatistics` behave similar as `387` (https://github.com/trinodb/trino/pull/13502)

**Comment by findepi** [comment]
> (X) No release notes entries required.

I think this should reduce metastore communication overhead and number of calls made to it.

**Comment by dain** [comment]
We're going to go with the alternate implementation in #13610

**Pull request closed without merging**